packages/iTwinUI-react/src/core/ComboBox/ComboBoxMenu.tsx

```diff
@@ -56,31 +56,27 @@ const VirtualizedComboBoxMenu = React.forwardRef(
       scrollToIndex: focusedVisibleIndex,
     });
 
-    const overflowY = getWindow()?.CSS?.supports?.('overflow-x: overlay')
-      ? { overflowY: 'overlay' }
-      : { overflowY: 'auto' };
+    const isOverflowOverlaySupported =
+      getWindow()?.CSS?.supports?.('overflow: overlay');
 
-    const styles = React.useMemo(
-      () => ({
-        minWidth,
-        maxWidth: `min(${minWidth * 2}px, 90vw)`,
-      }),
-      [minWidth],
-    );
+    const surfaceStyles = {
+      minWidth,
+      maxWidth: `min(${minWidth * 2}px, 90vw)`,
+      // max-height must be on the outermost element for virtual scroll
+      maxHeight: 'calc((var(--iui-component-height) - 1px) * 8.5)',
+      overflowY: isOverflowOverlaySupported ? 'overlay' : 'auto',
+      ...style,
+    } as React.CSSProperties;
 
     return (
-      <Surface
-        elevation={1}
-        style={{ ...styles, ...(overflowY as React.CSSProperties), ...style }}
-        {...rest}
-      >
+      <Surface style={surfaceStyles} {...rest}>
         <div {...outerProps}>
           <Menu
             id={`${id}-list`}
             setFocus={false}
             role='listbox'
             ref={mergeRefs(menuRef, innerProps.ref, forwardedRef)}
-            className={cx('iui-scroll', className)}
+            className={className}
```

Comment:
Just curious, would still using the `.iui-scroll` for `overlay-y` but just overriding the `max-height` remove the need for `isOverflowOverlaySupported`?

https://github.com/iTwin/iTwinUI/blob/9912acf9a0a526ccf34d1089ef03722ed26d5c62/packages/itwinui-css/src/menu/menu.scss#L30-L34